### PR TITLE
feature-<#5631>-<Create SpreadExecutionModel>

### PR DESCRIPTION
**Pull request by 31620000**
#### Description
Creating a spread execution model to execute when spread is desirably small in python.

#### Related Issue
#5631

#### Motivation and Context
To provide easy implementation of execution through tight spread to reduce slippage

#### Requires Documentation Change
Add desciption on this class in "Execution" page

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactor (non-breaking change which improves implementation)
- [ ] Performance (non-breaking change which improves performance. Please add associated performance test and results)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Non-functional change (xml comments/documentation/etc)

#### Checklist:

- [X] My code follows the code style of this project.
- [X] I have read the **CONTRIBUTING** [document](https://github.com/QuantConnect/Lean/blob/master/CONTRIBUTING.md).
- [X] I have added tests to cover my changes. 
- [ ] All new and existing tests passed.
- [X] My branch follows the naming convention `bug-<issue#>-<description>` or `feature-<issue#>-<description>`

### CODE REVIEW

**Comment by 31620000** [comment]
The tests failed by LiveOptionChainFromFuture something like that. It is unrelated to the execution model here itself, I'm confused on passing the tests.

File: Algorithm.Framework/Execution/SpreadExecutionModel.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+            for target in self.targetsCollection.OrderByMarginImpact(algorithm):
+                symbol = target.Symbol
+                
+                # calculate remaining quantity to be ordered
+                unorderedQuantity = OrderSizing.GetUnorderedQuantity(algorithm, target)
+                
+                # get pricing information
+                lastPrice = algorithm.Securities[symbol].Price
+                bidPrice = algorithm.Securities[symbol].BidPrice 
+                askPrice = algorithm.Securities[symbol].AskPrice
```
Comment (Martin-Molinero) [inline Algorithm.Framework/Execution/SpreadExecutionModel.py:63]: Might be better to store `algorithm.Securities[symbol]` once at the top and re use the result, avoid the dictionary access and py->C# multiple times. Same for the C# implementation 👍

**Review by Martin-Molinero — COMMENTED** [review]
Hey @31620000! Thanks for the PR ✨ 
Suggest creating a single PR for both C# & Py model. Unit tests should be added for these new models, there are plenty of examples available at https://github.com/QuantConnect/Lean/tree/master/Tests/Algorithm/Framework/Execution

File: Algorithm.Framework/Execution/SpreadExecutionModel.py (hunk 1 above)
Comment (31620000) [reply Algorithm.Framework/Execution/SpreadExecutionModel.py]: Thanks for the suggestion, I will make amendments.

Comment (31620000) [reply Algorithm.Framework/Execution/SpreadExecutionModel.py]: Hi Matin

please check on the test. I have no idea on how test error on LiveOptionChainProviderReturnsNoDataForOldFuture

**Review by Martin-Molinero — APPROVED** [review]
Hey @31620000! Thanks for the updates
I've pushed a commit with some minor tweaks, can ignore `LiveOptionChain` unit test failing, already skipped in master 👍

**Pull request merged**